tests/Output/StreamOutputTest.php

```diff
@@ -54,6 +55,20 @@ public function testReadingLineReadsFromInputStream(): void
         $this->assertEquals('foo', $this->output->readLine());
     }
 
+    public function testReadingLineThrowsRuntimeException(): void
+    {
+        $inputStream = fopen(__DIR__ . '/../../composer.json', 'r');
+        $outputStream = fopen(__DIR__ . '/../../composer.json', 'r');
+        $output = new StreamOutput($outputStream, $inputStream, $this->compiler);
+        try {
+            $output->readLine();
+        } catch (RuntimeException $e) {
+            $this->assertSame('Failed to read line', $e->getMessage());
+            fclose($inputStream);
```

Comment:
Put these two `fclose` into a `finally` block just in the case that `$output->readLine()` does not throw the expected `RuntimeException`.